cpp/src/arrow/util/decimal.h

```diff
@@ -134,6 +135,20 @@ class ARROW_EXPORT Decimal128 {
   /// \brief Convert Decimal128 from one scale to another
   Status Rescale(int32_t original_scale, int32_t new_scale, Decimal128* out) const;
 
+  template <typename T, typename = EnableIfIsOneOf<T, int32_t, int64_t>>
```

Comment:
Seems to me the `EnableIfIsOneOf` restriction isn't useful and removing it would actually fix the compile errors.